io/src/error.rs

```diff
@@ -103,16 +103,17 @@ impl From<Error> for std::io::Error {
 }
 
 macro_rules! define_errorkind {
-    ($($kind: ident),*) => {
-        #[non_exhaustive]
+    ($($kind: ident, $doc: expr);*) => {
```

Comment:
I suggest this instead:

```rust
($($(#[$($attr:tt)*])* $kind:ident),*) => {
    // ...
    pub enum ErrorKind {
        $(
            $(#[$($attr)*])*
            $kind
        )*
    }
}
```

So that you can use normal-looking  syntax:

```rust
define_errorkind! {
    /// An entity was not found, often a file.
    NotFound,
    // ...
}
```

Reply:
Agreed. I think it's worth blocking the release on this.

Reply:
Oh cool, that's way better. I'll fix the place I copied this from too - review for the win.

Reply:
I copied from `opcodes::all_opcodes` but to change that is a massive changeset that I can't be bothered doing right now, too much other more important stuff to do.